bin/ch-test

```diff
@@ -169,12 +320,12 @@ win () {
     printf "\nAll tests passed.\n"
 }
 
-
-top_dir=$(readlink -f "${ch_bin%/*}")
+bin=$(readlink -f "$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)")
+topdir=$(readlink -f "${bin%/*}")
```

Comment:
(1) I would set these up at the top of the script, since that's where the others do it.
(2) Recommend rename `topdir` to `ch_base` for consistency (or even add it to `base.sh`).
(3) Why `readlink`?

Reply:
Guarantee absolute path. If ch-test is sitting in some bin directory that is a symlink to somewhere else, I want to refer to that somewhere else directly. Not sure my preference for absolute path is justifiable (it very well could be hand-wavy), it just _feels_ right.

Reply:
I think you do not need it in that case. "Guarantee absolute path" is going to be rarely needed.